tests/components/tomorrowio/test_init.py

```diff
@@ -66,22 +71,49 @@ async def test_update_intervals(hass: HomeAssistant) -> None:
     assert hass.data[DOMAIN][
         config_entry.data[CONF_API_KEY]
     ].update_interval == timedelta(minutes=32)
+    assert len(tomorrowio_config_entry_update.call_args_list) == 1
 
-    # Adding a second config entry should cause the update interval to double
-    config_entry_2 = MockConfigEntry(
-        domain=DOMAIN,
-        data=data,
-        options={CONF_TIMESTEP: 1},
-        unique_id=f"{_get_unique_id(hass, data)}_1",
-        version=1,
-    )
-    config_entry_2.add_to_hass(hass)
-    assert await hass.config_entries.async_setup(config_entry_2.entry_id)
+    # 30 minutes in, no updates yet
+    async_fire_time_changed(hass, now + timedelta(minutes=30))
     await hass.async_block_till_done()
-    assert config_entry.data[CONF_API_KEY] == config_entry_2.data[CONF_API_KEY]
-    assert hass.data[DOMAIN][
-        config_entry.data[CONF_API_KEY]
-    ].update_interval == timedelta(minutes=64)
+    assert len(tomorrowio_config_entry_update.call_args_list) == 1
+
+    # 32 minutes in, we get a new update
+    async_fire_time_changed(hass, now + timedelta(minutes=32))
+    await hass.async_block_till_done()
+    assert len(tomorrowio_config_entry_update.call_args_list) == 2
+
+    with patch(
+        "homeassistant.helpers.update_coordinator.utcnow",
+        return_value=now + timedelta(minutes=32),
+    ):
+        # Adding a second config entry should cause the update interval to double
+        config_entry_2 = MockConfigEntry(
+            domain=DOMAIN,
+            data=data,
+            options={CONF_TIMESTEP: 1},
+            unique_id=f"{_get_unique_id(hass, data)}_1",
+            version=1,
+        )
+        config_entry_2.add_to_hass(hass)
+        assert await hass.config_entries.async_setup(config_entry_2.entry_id)
+        await hass.async_block_till_done()
+        assert config_entry.data[CONF_API_KEY] == config_entry_2.data[CONF_API_KEY]
+        assert hass.data[DOMAIN][
+            config_entry.data[CONF_API_KEY]
+        ].update_interval == timedelta(minutes=64)
```

Comment:
We want to avoid asserting integration details like this.

Reply:
Isn't that what I'm effectively doing with moving times forward? I can get rid of these assertions but I'm using these numbers to validate the number of API calls anyway

Reply:
Time moving forward isn't a detail. Library api calls isn't a detail.